src/core/server/http_resources/get_apm_config.ts

```diff
@@ -6,12 +6,18 @@
  * Side Public License, v 1.
  */
 
-import agent from 'elastic-apm-node';
+import agent, { AgentConfigOptions } from 'elastic-apm-node';
 import { getConfiguration } from '@kbn/apm-config-loader';
 
+function shouldInstrumentClient(config?: AgentConfigOptions): boolean {
```

Comment:
maybe this logic belongs to `'@kbn/apm-config-loader'`. I left it here as it was implemented here before.

Reply:
First, I don't think you should block on me. I don't have a feel for Kibana code structure at all.

I think having all logic for APM config in `@kbn/apm-config-loader` might be less surprising.